tests/unittests/test_dispatcher.py

```diff
@@ -0,0 +1,95 @@
+        by default
+        '''
+        ctrl = testutils.start_mockhost(script_root=self.dispatcher_funcs_dir)
+
+        async with ctrl as host:
+            # Ensure the function can be properly loaded
+            func_id, load_r = await host.load_function('show_context')
+            self.assertEqual(load_r.response.function_id, func_id)
+            self.assertEqual(load_r.response.result.status,
+                             protos.StatusResult.Success)
+
+            # Ensure the function can be properly invoked
+            invoke_id, call_r = await host.invoke_function(
+                'show_context', [
+                    protos.ParameterBinding(
+                        name='req',
+                        data=protos.TypedData(
+                            http=protos.RpcHttp(
+                                method='GET'
+                            )
+                        )
+                    )
+                ])
+            self.assertIsNotNone(invoke_id)
+            self.assertEqual(call_r.response.result.status,
+                             protos.StatusResult.Success)
+
+            # Ensure the dispatcher sync threadpool count is set to 1
+            self.assertEqual(ctrl._worker._sync_tp_max_workers, 1)
+
+    async def test_dispatcher_sync_threadpool_set_worker(self):
+        '''Test if the sync threadpool maximum worker can be set
+        '''
+        # Configure thread pool max worker
+        os.environ.update({PYTHON_THREADPOOL_THREAD_COUNT: '5'})
+        ctrl = testutils.start_mockhost(script_root=self.dispatcher_funcs_dir)
+
+        async with ctrl as host:
+            # Ensure the function can be properly loaded
+            func_id, load_r = await host.load_function('show_context')
+            self.assertEqual(load_r.response.function_id, func_id)
+            self.assertEqual(load_r.response.result.status,
+                             protos.StatusResult.Success)
+
+            # Ensure the function can be properly invoked
+            invoke_id, call_r = await host.invoke_function(
+                'show_context', [
+                    protos.ParameterBinding(
+                        name='req',
+                        data=protos.TypedData(
+                            http=protos.RpcHttp(
+                                method='GET'
+                            )
+                        )
+                    )
+                ])
+            self.assertIsNotNone(invoke_id)
+            self.assertEqual(call_r.response.result.status,
+                             protos.StatusResult.Success)
+
+            # Ensure the dispatcher sync threadpool count is set to 1
+            self.assertEqual(ctrl._worker._sync_tp_max_workers, 5)
+
+    async def test_dispatcher_sync_threadpool_invalid_worker(self):
+        '''Test when sync threadpool maximum worker is set to an invalid value,
+        the host should fail to start
```

Comment:
Do we want to fail or do we want to default to 1? I prefer not failing but making it clear to the customer when running functions that there is an invalid value for the Application Settings.

@anirudhgarg - Any thoughts?

Reply:
This is related to my previous comment - 

> If the customer puts in an incorrect value, it might throw an exception when casting to int - you will need to take care of that - try-except, etc.

Reply:
OK, I will add a validator field in get_app_setting